src/lib/apputils/net-server.c

```diff
@@ -753,7 +764,7 @@ setup_socket(struct socksetup *data, struct bind_address *ba,
     }
 
     /* Try to turn on pktinfo for UDP wildcard sockets. */
-    if (ba->type == UDP && ba->address == NULL) {
+    if (ba->type == UDP && is_wildcard(sock_address)) {
         krb5_klog_syslog(LOG_DEBUG, _("Setting pktinfo on socket %s"),
```

Comment:
Thanks for testing.

For the purpose of a backported fix, I would like to avoid using pktinfo for non-wildcard sockets, as we have never done that.  (Prior to 1.15, we either used pktinfo on a wildcard socket or iterated over the interfaces and bound to each interface address without pktinfo.)  We can consider using pktinfo everywhere for 1.16, although my memory agrees with Sarah's that there is a portability issue lurking there.